### PR TITLE
Support count with collation in MongoDB with Panache

### LINKED ISSUE #27063: Panache MongoDB doesn't support collation in count
### Describe the bug

Panache MongoDB doesn't support `collation` for `count`.
Taking a look in the `io.quarkus.mongodb.panache.common.runtime.CommonPanacheQueryImpl` and `io.quarkus.mongodb.panache.common.reactive.runtime.CommonReactivePanacheQueryImpl` we can see the method `count` doesn't support `collation`.

Native queries and panache queries already support this, we can confirm in this [PR](https://github.com/loicmathieu/quarkus/commit/5c0a0a2112cc2d33ecc300618f0694846d0f0f1c#diff-610ce4a20cb8ce7c6942db761ecaebe7a4b6fc761e4b4f0b15b7406ec7db305f) from @loicmathieu already merged.

I implemented the fix, basically I changed a small piece in `CommonReactivePanacheQueryImpl` and `CommonPanacheQueryImpl`:

```
public Uni<Long> count() {
        if (count == null) {
            CountOptions countOptions = new CountOptions();
            if (collation != null) {
                countOptions.collation(collation);
            }

            count = mongoQuery == null
                    ? collection.countDocuments()
                    : collection.countDocuments(mongoQuery, countOptions);
        }
        return count;
    }
```

```
public long count() {
        if (count == null) {
            Bson query = getQuery();
            CountOptions countOptions = new CountOptions();
            if (collation != null) {
                countOptions.collation(collation);
            }

            count = clientSession == null
                    ? collection.countDocuments(query, countOptions)
                    : collection.countDocuments(clientSession, query, countOptions);
        }
        return count;
    }
```

Here is the [PR](https://github.com/quarkusio/quarkus/pull/27062), let me know if I need to change something.

Best regards.

### Expected behavior

_No response_

### Actual behavior

_No response_

### How to Reproduce?

_No response_

### Output of `uname -a` or `ver`

_No response_

### Output of `java -version`

_No response_

### GraalVM version (if different from Java)

_No response_

### Quarkus version or git rev

_No response_

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by diogocarleto**
This is a fix to allow us execute `count` queries with collation

Fixes #27063

### CODE REVIEW

**Review by loicmathieu — APPROVED** [review]
This is pretty straigthforward regarding what I did with the find query.
Looks good to me, thanks.

**Comment by gsmet** [comment]
I force pushed to squash the commits. Let's wait for CI before merging.

**Review by evanchooly — APPROVED** [review]

**Pull request merged**
